### ISSUE TITLE
Exception if the arlo's api is very slow

### ISSUE THREAD

**Issue by maheus** [twrecked/hass-aarlo#57]
reguraly the arlo's api is very slow and causes exceptions 

```
May 12 14:58:35 hassbian hass[20354]: Exception in thread ArloBackgroundWorker:
May 12 14:58:35 hassbian hass[20354]: Traceback (most recent call last):
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 600, in urlopen
May 12 14:58:35 hassbian hass[20354]:     chunked=chunked)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 384, in _make_request
May 12 14:58:35 hassbian hass[20354]:     six.raise_from(e, None)
May 12 14:58:35 hassbian hass[20354]:   File "<string>", line 2, in raise_from
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 380, in _make_request
May 12 14:58:35 hassbian hass[20354]:     httplib_response = conn.getresponse()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 1198, in getresponse
May 12 14:58:35 hassbian hass[20354]:     response.begin()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 297, in begin
May 12 14:58:35 hassbian hass[20354]:     version, status, reason = self._read_status()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 266, in _read_status
May 12 14:58:35 hassbian hass[20354]:     raise RemoteDisconnected("Remote end closed connection without"
May 12 14:58:35 hassbian hass[20354]: http.client.RemoteDisconnected: Remote end closed connection without response
May 12 14:58:35 hassbian hass[20354]: During handling of the above exception, another exception occurred:
May 12 14:58:35 hassbian hass[20354]: Traceback (most recent call last):
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/adapters.py", line 449, in send
May 12 14:58:35 hassbian hass[20354]:     timeout=timeout
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 638, in urlopen
May 12 14:58:35 hassbian hass[20354]:     _stacktrace=sys.exc_info()[2])
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/util/retry.py", line 367, in increment
May 12 14:58:35 hassbian hass[20354]:     raise six.reraise(type(error), error, _stacktrace)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/packages/six.py", line 685, in reraise
May 12 14:58:35 hassbian hass[20354]:     raise value.with_traceback(tb)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 600, in urlopen
May 12 14:58:35 hassbian hass[20354]:     chunked=chunked)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 384, in _make_request
May 12 14:58:35 hassbian hass[20354]:     six.raise_from(e, None)
May 12 14:58:35 hassbian hass[20354]:   File "<string>", line 2, in raise_from
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/urllib3/connectionpool.py", line 380, in _make_request
May 12 14:58:35 hassbian hass[20354]:     httplib_response = conn.getresponse()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 1198, in getresponse
May 12 14:58:35 hassbian hass[20354]:     response.begin()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 297, in begin
May 12 14:58:35 hassbian hass[20354]:     version, status, reason = self._read_status()
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/http/client.py", line 266, in _read_status
May 12 14:58:35 hassbian hass[20354]:     raise RemoteDisconnected("Remote end closed connection without"
May 12 14:58:35 hassbian hass[20354]: urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
May 12 14:58:35 hassbian hass[20354]: During handling of the above exception, another exception occurred:
May 12 14:58:35 hassbian hass[20354]: Traceback (most recent call last):
May 12 14:58:35 hassbian hass[20354]:   File "/home/homeassistant/.homeassistant/custom_components/aarlo/pyaarlo/background.py", line 58, in run
May 12 14:58:35 hassbian hass[20354]:     timeout = self._run_next()
May 12 14:58:35 hassbian hass[20354]:   File "/home/homeassistant/.homeassistant/custom_components/aarlo/pyaarlo/background.py", line 33, in _run_next
May 12 14:58:35 hassbian hass[20354]:     job['callback']( **job['args'] )
May 12 14:58:35 hassbian hass[20354]:   File "/home/homeassistant/.homeassistant/custom_components/aarlo/pyaarlo/camera.py", line 109, in _update_last_image
May 12 14:58:35 hassbian hass[20354]:     img = http_get( url )
May 12 14:58:35 hassbian hass[20354]:   File "/home/homeassistant/.homeassistant/custom_components/aarlo/pyaarlo/util.py", line 26, in http_get
May 12 14:58:35 hassbian hass[20354]:     ret = requests.get(url)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/api.py", line 75, in get
May 12 14:58:35 hassbian hass[20354]:     return request('get', url, params=params, **kwargs)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/api.py", line 60, in request
May 12 14:58:35 hassbian hass[20354]:     return session.request(method=method, url=url, **kwargs)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/sessions.py", line 533, in request
May 12 14:58:35 hassbian hass[20354]:     resp = self.send(prep, **send_kwargs)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/sessions.py", line 646, in send
May 12 14:58:35 hassbian hass[20354]:     r = adapter.send(request, **kwargs)
May 12 14:58:35 hassbian hass[20354]:   File "/srv/homeassistant/lib/python3.5/site-packages/requests/adapters.py", line 498, in send
May 12 14:58:35 hassbian hass[20354]:     raise ConnectionError(err, request=request)
May 12 14:58:35 hassbian hass[20354]: requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
May 12 14:58:35 hassbian hass[20354]: During handling of the above exception, another exception occurred:
May 12 14:58:35 hassbian hass[20354]: Traceback (most recent call last):
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
May 12 14:58:35 hassbian hass[20354]:     self.run()
May 12 14:58:35 hassbian hass[20354]:   File "/home/homeassistant/.homeassistant/custom_components/aarlo/pyaarlo/background.py", line 63, in run
May 12 14:58:35 hassbian hass[20354]:     self._lock.wait( timeout - now )
May 12 14:58:35 hassbian hass[20354]:   File "/usr/lib/python3.5/threading.py", line 241, in __exit__
May 12 14:58:35 hassbian hass[20354]:     return self._lock.__exit__(*args)
May 12 14:58:35 hassbian hass[20354]: RuntimeError: cannot release un-acquired lock


```

and after that , arlo's plugin is blocked and a restart of homeassistant is necessary

**Comment by twrecked** [comment]
Thanks. I think the fix is relatively straight forward.

For testing the fix, is it easily reproducible?

**Comment by twrecked** [comment]
Can you try the `timeout-exception` branch (https://github.com/twrecked/hass-aarlo/tree/timeout-exception) it should handle general exceptions better.

**Comment by maheus** [comment]
I think it is very hard to reproducible but the arlo's platform is regulary slow so i will put in place your branch and i will check the logs every days ;)

**Comment by maheus** [comment]
branch in place

**Comment by twrecked** [comment]
Thanks. I'll merge the fix in anyway this weekend - it should only make things better - but it would be nice if we could catch it fixing something.

**Comment by twrecked** [comment]
Fixed in #60
